webview/src/plots/components/templatePlots/TemplatePlotsGrid.tsx

```diff
@@ -43,11 +43,47 @@ export const TemplatePlotsGrid: React.FC<TemplatePlotsGridProps> = ({
   parentDraggedOver
 }) => {
   const [order, setOrder] = useState<string[]>([])
+  const [disabledDrag, setDisabledDrag] = useState('')
 
   useEffect(() => {
     setOrder(entries.map(({ id }) => id))
   }, [entries])
 
+  useEffect(() => {
+    const panels = document.querySelectorAll('.vega-bindings')
+    const addDisabled = (e: Event) => {
+      setDisabledDrag(
+        (e.currentTarget as HTMLFormElement).parentElement?.parentElement
+          ?.parentElement?.id || ''
+      )
+    }
+
+    const removeDisabled = () => {
+      setDisabledDrag('')
+    }
+
+    const disableClick = (e: Event) => {
+      e.stopPropagation()
+    }
+
+    for (const panel of Object.values(panels)) {
+      panel.removeEventListener('mouseenter', addDisabled)
+      panel.removeEventListener('mouseleave', removeDisabled)
+      panel.removeEventListener('click', disableClick)
+      panel.addEventListener('mouseenter', addDisabled)
+      panel.addEventListener('mouseleave', removeDisabled)
+      panel.addEventListener('click', disableClick)
```

Comment:
[Q] What is the reason for recreating these?

Reply:
Since it's being created by vega the `vega-bindings` and that the `DragDropContainer` rebuilds the plots, setting these events on re-render is the only way to hook on them. Doing so once would create them too early. Maybe I can set up an observer or something to add the listeners once the elements are ready. I'll look into it and do a follow up if possible.